src/test/java/de/tum/in/www1/artemis/authorization/AuthorizationTest.java

```diff
@@ -0,0 +1,156 @@
+package de.tum.in.www1.artemis.authorization;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.mvc.method.RequestMappingInfo;
+import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationBambooBitbucketJiraTest;
+
+/**
+ * Contains the one automatic test covering all rest endpoints for authorization tests.
+ */
+public class AuthorizationTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Autowired
+    private AuthorizationTestService authorizationTestService;
+
+    @Test
+    public void testEndpoints() throws InvocationTargetException, IllegalAccessException {
+        var requestMappingHandlerMapping = applicationContext.getBean("requestMappingHandlerMapping", RequestMappingHandlerMapping.class);
+        Map<RequestMappingInfo, HandlerMethod> map = requestMappingHandlerMapping.getHandlerMethods();
+        // Only search our Artemis endpoints
+        map = map.entrySet().stream().filter(entry -> entry.getValue().getBeanType().getPackage().getName().startsWith("de.tum.in.www1.artemis"))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        Map<Class<?>, List<String>> classReports = new HashMap<>();
+        Map<Method, List<String>> methodReports = new HashMap<>();
+
+        // Test each endpoint and collect the reports
+        for (Map.Entry<RequestMappingInfo, HandlerMethod> entry : map.entrySet()) {
+            testEndpoint(entry.getKey(), entry.getValue(), classReports, methodReports);
+        }
+
+        // Process the reports for better readability
+        Map<Class<?>, List<String>> distinctClassReports = classReports.entrySet().stream().peek(entry -> entry.setValue(entry.getValue().stream().distinct().toList()))
```

Comment:
Would it make sense to save the class reports as `Set<String>` (maybe even from the beginning in line 40)? Then it would be enough to `new HashSet<>(entry.getValue())` (or not necessary at all).
```suggestion
        Map<Class<?>, List<String>> distinctClassReports = classReports.entrySet().stream().map(entry -> new Map.Entry<>(entry.getKey(), new HashSet<>(entry.getValue())))
```

Reply:
I don't even need to create a new Set as the Set itself doesn't allow any duplicates. Good catch.